gluten-core/src/main/resources/substrait/proto/substrait/algebra.proto

```diff
@@ -121,6 +121,12 @@ message ReadRel {
       // The index of the partition this item belongs to
       uint64 partition_index = 6;
 
+      repeated partitionColumn partition_columns = 16;
+      message partitionColumn {
+        string key = 1;
+        string value = 2;
+      }
```

Comment:
Please avoid duplicate definition with [PartitionColumns](https://github.com/oap-project/gluten/blob/main/gluten-core/src/main/resources/substrait/proto/substrait/type.proto#L239-L244), can we unify them together?

Reply:
Got it, will refactor later.

Reply:
This parts has been updated, could you review again?

Reply:
Thanks. Please update [SubstraitModifications.md](https://github.com/oap-project/gluten/blob/main/docs/developers/SubstraitModifications.md#modifications-to-typeproto) at the same time.